### PR TITLE
Minimal composer file

### PR DESCRIPTION
**Pull request by dannylamb**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/vendor/
```
Comment (DiegoPino) [inline .gitignore]: should we also ignoring composer.lock?

Comment (dannylamb) [inline .gitignore]: I think we've swapped sides on this issue.  Last time, you convinced me (rightly so, I feel) we should keep the composer.lock file: https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file

**Comment by DiegoPino** [comment]
@dannylamb true. 😄 I still thinke that on release versions, or at least semi-stable ones, `composer.lock ` should not be ignored so we can assure minimum stability/tested versions and happy to read you agree also! But since all is moving so fast here and i don't see a composer.lock in your pull, every time we do a `git pull` (after you have previously did a `composer update`) we have to basically stash everything, and then run `composer update` again. Which can be awful slow, weird, etc to test.  So the solution is this: true, not ignore composer.lock, but publish it! So, this pull should include composer.lock, or avoid it all ignoring it. Does it make sense?

File: .gitignore (hunk 1 above)
Comment (DiegoPino) [inline .gitignore]: Solution is to publish a composer.lock or ignore it. I added a comment under this one to explain myself better.

**Comment by dannylamb** [comment]
Publishing the lockfile and not ignoring it is the way to go.  Give me a moment.

**Comment by dannylamb** [comment]
Funnily enough, since there's no dependencies defined in the 'require' block, it doesn't generate a .lock file.  We'll just have to make sure to commit one if a dependency gets added.

**Comment by DiegoPino** [comment]
@dannylamb funny! I would had assumed it would state that it depends on drupal at least. Any idea on how to make that?  reading this https://www.drupal.org/docs/develop/using-composer/using-composer-to-manage-drupal-site-dependencies but not sure how to make it fit. The module is fully dependant on drupal 8 and we should also make sure here we are targeting the "less unstable" version we are sure it works with right?

**Pull request merged**